multinet/api/tests/conftest.py

```diff
@@ -11,10 +20,94 @@ def api_client() -> APIClient:
 
 
 @pytest.fixture
-def authenticated_api_client(user) -> APIClient:
+def authenticated_api_client(user: User) -> APIClient:
     client = APIClient()
     client.force_authenticate(user=user)
     return client
 
 
+@pytest.fixture
+def owned_workspace(user: User, workspace: Workspace) -> Workspace:
+    """Return a workspace with the `user` fixture as an owner."""
+    assign_perm('owner', user, workspace)
+
+    return workspace
+
+
+@pytest.fixture
+def populated_node_table(owned_workspace: Workspace) -> Table:
+    nodes = generate_arango_documents(5)
+    table, created = Table.objects.get_or_create(
+        name=Faker().pystr(), edge=False, workspace=owned_workspace
+    )
+
+    if created:
+        table.save()
+
+    table.put_rows(nodes)
+    return table
+
+
+@pytest.fixture
+def populated_edge_table(owned_workspace: Workspace, populated_node_table: Table) -> Table:
+    table, created = Table.objects.get_or_create(
+        name=Faker().pystr(), edge=True, workspace=owned_workspace
+    )
+
+    if created:
+        table.save()
+
+    nodes = list(populated_node_table.get_rows())
+    edges = [{'_from': a['_id'], '_to': b['_id']} for a, b in itertools.combinations(nodes, 2)]
+    table.put_rows(edges)
+
+    return table
+
+
+@pytest.fixture
+def populated_network(owned_workspace: Workspace, populated_edge_table: Table) -> Network:
+    node_tables = list(populated_edge_table.find_referenced_node_tables().keys())
+    network_name = Faker().pystr()
+
+    # Create graph in arango before creating the Network object here
+    owned_workspace.get_arango_db().create_graph(
+        network_name,
+        edge_definitions=[
+            {
+                'edge_collection': populated_edge_table.name,
+                'from_vertex_collections': node_tables,
+                'to_vertex_collections': node_tables,
+            }
+        ],
+    )
+
+    network, created = Network.objects.get_or_create(
```

Comment:
What's with the conditional logic in a fixture? Seems to me this should either always get an existing object or create a new one deterministically.

Reply:
Are you saying I should just change this to always create a new Network? That's what's supposed to happen here, but I guess I was trying to guard against some obscure collision. These workspaces are created when the tests begin though, so I suppose it's not really an issue.

Reply:
Yes, let's do that. If collisions happen, we can figure out how to avoid them.